File: chapter21/04_bevy-game/create_assets.py
#!/usr/bin/env python3
"""
Create simple game assets for the Bevy game example.
This script creates a simple player sprite as a PNG file.
"""

from PIL import Image, ImageDraw
import os

def create_player_sprite():
    """Create a simple player sprite (blue square with white border)"""
    size = 64
    img = Image.new('RGBA', (size, size), (0, 0, 0, 0))  # Transparent background
    draw = ImageDraw.Draw(img)
    
    # Draw blue square with white border
    draw.rectangle([4, 4, size-4, size-4], fill=(0, 100, 255, 255), outline=(255, 255, 255, 255), width=2)
    
    # Add a simple face
    # Eyes
    draw.ellipse([16, 20, 20, 24], fill=(255, 255, 255, 255))
    draw.ellipse([44, 20, 48, 24], fill=(255, 255, 255, 255))
    
    # Smile
    draw.arc([20, 35, 44, 50], start=0, end=180, fill=(255, 255, 255, 255), width=2)
    
    return img

def create_background():
    """Create a simple background texture"""
    size = 128
    img = Image.new('RGB', (size, size), (50, 150, 50))  # Green background
    draw = ImageDraw.Draw(img)
    
    # Add some grass-like texture
    for i in range(0, size, 16):
        for j in range(0, size, 16):
            if (i + j) % 32 == 0:
                draw.rectangle([i, j, i+8, j+8], fill=(40, 120, 40))
    
    return img

def main():
    # Create assets directory if it doesn't exist
    assets_dir = 'assets'
    if not os.path.exists(assets_dir):
        os.makedirs(assets_dir)
    
    print("🎨 Creating game assets...")
    
    # Create player sprite
    player_img = create_player_sprite()
    player_path = os.path.join(assets_dir, 'player.png')
    player_img.save(player_path)
    print(f"✅ Created player sprite: {player_path}")
    
    # Create background
    bg_img = create_background()
    bg_path = os.path.join(assets_dir, 'background.png')
    bg_img.save(bg_path)
    print(f"✅ Created background: {bg_path}")
    
    print("\n🎮 Assets created successfully!")
    print("You can now run the game with: cargo run")

if __name__ == "__main__":
    try:
        main()
    except ImportError:
        print("❌ PIL (Pillow) is required to create assets.")
        print("Install it with: pip install Pillow")
        print("\nAlternatively, you can:")
        print("1. Create a 64x64 PNG file named 'player.png' in the assets/ folder")
        print("2. Use any image editing software or download a sprite online")
        print("3. Run: cargo run")

File: chapter21/04_bevy-game/create_assets.js
#!/usr/bin/env node
/**
 * Create simple game assets for the Bevy game example using Node.js Canvas.
 * This script creates a simple player sprite as a PNG file.
 */

const fs = require('fs');
const path = require('path');

// Simple canvas implementation for creating basic sprites
function createSimpleSprite() {
    // Create a simple SVG that can be converted to PNG
    const svgContent = `
<svg width="64" height="64" xmlns="http://www.w3.org/2000/svg">
  <!-- Blue square with white border -->
  <rect x="4" y="4" width="56" height="56" fill="#0064ff" stroke="#ffffff" stroke-width="2"/>
  
  <!-- Simple face -->
  <!-- Eyes -->
  <circle cx="20" cy="22" r="3" fill="#ffffff"/>
  <circle cx="44" cy="22" r="3" fill="#ffffff"/>
  
  <!-- Smile -->
  <path d="M 20 35 Q 32 45 44 35" stroke="#ffffff" stroke-width="2" fill="none"/>
</svg>`;
    
    return svgContent;
}

function createBackground() {
    const svgContent = `
<svg width="128" height="128" xmlns="http://www.w3.org/2000/svg">
  <!-- Green background -->
  <rect width="128" height="128" fill="#329632"/>
  
  <!-- Grass pattern -->
  <defs>
    <pattern id="grass" x="0" y="0" width="16" height="16" patternUnits="userSpaceOnUse">
      <rect width="16" height="16" fill="#329632"/>
      <rect x="0" y="0" width="8" height="8" fill="#287828"/>
    </pattern>
  </defs>
  <rect width="128" height="128" fill="url(#grass)"/>
</svg>`;
    
    return svgContent;
}

function main() {
    // Create assets directory if it doesn't exist
    const assetsDir = 'assets';
    if (!fs.existsSync(assetsDir)) {
        fs.mkdirSync(assetsDir);
    }
    
    console.log('🎨 Creating game assets...');
    
    // Create player sprite (as SVG for now)
    const playerSvg = createSimpleSprite();
    const playerPath = path.join(assetsDir, 'player.svg');
    fs.writeFileSync(playerPath, playerSvg);
    console.log(`✅ Created player sprite: ${playerPath}`);
    
    // Create background
    const bgSvg = createBackground();
    const bgPath = path.join(assetsDir, 'background.svg');
    fs.writeFileSync(bgPath, bgSvg);
    console.log(`✅ Created background: ${bgPath}`);
    
    // Create a simple fallback PNG using text representation
    const fallbackInstructions = `
# Bevy Game Assets

Since we created SVG files, you have a few options:

## Option 1: Convert SVG to PNG
Use an online converter or tool like Inkscape:
- Convert player.svg to player.png
- Convert background.svg to background.png

## Option 2: Use any 64x64 PNG image
- Download a simple sprite from the internet
- Create one in any image editor
- Name it 'player.png' and put it in the assets/ folder

## Option 3: Use the built-in Bevy shapes
The game will still run, it just won't show the sprite if the PNG is missing.

Run the game with: cargo run
`;
    
    fs.writeFileSync(path.join(assetsDir, 'README.txt'), fallbackInstructions);
    
    console.log('\n🎮 Assets created successfully!');
    console.log('Note: SVG files created. Convert to PNG or use any 64x64 image as player.png');
    console.log('You can now run the game with: cargo run');
}

if (require.main === module) {
    main();
}